### PR TITLE
DM-7326:  Adapt FireflyClient to Python 3.

### PR DESCRIPTION
**Pull request by cwang2016**
- Made FireflyClient to be Python 2/3 compatible per instructions of 
   https://sqr-014.lsst.io/v/DM-6315/index.html
- change FireflyClient.py to be firefly_client.py

### CODE REVIEW

File: src/firefly/python/display/firefly_client.py [hunk 1]
```diff
@@ -157,7 +161,7 @@ def __init__(self, host=_my_localhost, channel=None):
         self.connect()
 
     def _handle_event(self, ev):
-        for callback, eventIDList in self.listeners.items():
+        for callback, eventIDList in list(self.listeners.items()):
```
Comment (stargaser) [inline src/firefly/python/display/firefly_client.py]: Here is a case where `futurize -2` is too conservative by wrapping  `self.listeners.items()` in a list. The `list()` can be removed.

File: src/firefly/python/display/firefly_client.py [hunk 2]
```diff
@@ -239,7 +241,7 @@ def add_listener(self, callback, name=ALL):
 
         """
 
-        if callback not in self.listeners.keys():
+        if callback not in list(self.listeners.keys()):
```
Comment (stargaser) [inline src/firefly/python/display/firefly_client.py]: The `list()` can be removed. futurize is being overly cautious here.

File: src/firefly/python/display/firefly_client.py [hunk 3]
```diff
@@ -262,7 +264,7 @@ def remove_listener(self, callback, name=ALL):
         The callback listener is removed if all events are removed from the callback.
         """
 
-        if callback in self.listeners.keys():
+        if callback in list(self.listeners.keys()):
```
Comment (stargaser) [inline src/firefly/python/display/firefly_client.py]: `list()` should be removed.

File: src/firefly/python/display/firefly_client.py [hunk 4]
```diff
@@ -533,7 +537,8 @@ def show_fits(self, file_on_server=None, plot_id=None, **additional_params):
 
         additional_params : dict, optional
             Dictionary of any valid fits viewer plotting parameters,
-            see `fits plotting parameters<https://github.com/Caltech-IPAC/firefly/blob/dev/docs/fits-plotting-parameters.md>`_.
+            see `fits plotting
+            parameters<https://github.com/Caltech-IPAC/firefly/blob/dev/docs/fits-plotting-parameters.md>`_.
```
Comment (stargaser) [inline src/firefly/python/display/firefly_client.py]: Ooh, this link is extremely helpful! Like!

File: src/firefly/python/display/firefly_client.py [hunk 5]
```diff
@@ -648,7 +653,7 @@ def show_xyplot(self, file_on_server, **additional_params):
 
         url = self.url_root + "?cmd=pushXYPlot"
         if additional_params:
-            url += '&' + '&'.join(['%s=%s' % (k, v) for k, v in additional_params.items()])
+            url += '&' + '&'.join(['%s=%s' % (k, v) for k, v in list(additional_params.items())])
```
Comment (stargaser) [inline src/firefly/python/display/firefly_client.py]: The `list()` can be removed. futurize is being too conservative again.

File: src/firefly/python/display/firefly_client.py [hunk 6]
```diff
@@ -1,21 +1,25 @@
 """
-Module of FireflyClient.py
+Module of firefly_client.py
```
Comment (stargaser) [inline src/firefly/python/display/firefly_client.py]: Thank you for changing the name. This is definitely better.

**Comment by stargaser** [comment]
Review complete. Thank you for the cleanup and renaming. There are just a few instances where `futurize -2` was overly conservative in wrapping things in `list()` that should be removed.

**Comment by cwang2016** [comment]
It is great to resolve the puzzle on if applying list() to dict's keys(), items() and values().  Thanks.

**Pull request merged**